plugins/fs/src/commands.rs

```diff
@@ -1002,37 +1002,88 @@ pub fn resolve_path<R: Runtime>(
     let scope = tauri::scope::fs::Scope::new(
         webview,
         &FsScope::Scope {
-            allow: webview
-                .fs_scope()
-                .allowed
-                .lock()
-                .unwrap()
-                .clone()
-                .into_iter()
-                .chain(global_scope.allows().iter().filter_map(|e| e.path.clone()))
+            allow: global_scope
+                .allows()
+                .iter()
+                .filter_map(|e| e.path.clone())
                 .chain(command_scope.allows().iter().filter_map(|e| e.path.clone()))
                 .collect(),
-            deny: webview
-                .fs_scope()
-                .denied
-                .lock()
-                .unwrap()
-                .clone()
-                .into_iter()
-                .chain(global_scope.denies().iter().filter_map(|e| e.path.clone()))
+            deny: global_scope
+                .denies()
+                .iter()
+                .filter_map(|e| e.path.clone())
                 .chain(command_scope.denies().iter().filter_map(|e| e.path.clone()))
                 .collect(),
             require_literal_leading_dot: webview.fs_scope().require_literal_leading_dot,
         },
     )?;
 
-    if scope.is_allowed(&path) {
+    let fs_scope = webview.fs_scope();
+
+    let require_literal_leading_dot = fs_scope.require_literal_leading_dot.unwrap_or(cfg!(unix));
+
+    if fs_scope
+        .scope
+        .as_ref()
+        .map(|s| is_forbidden(s, &path, require_literal_leading_dot))
+        .unwrap_or(false)
+        || is_forbidden(&scope, &path, require_literal_leading_dot)
+    {
+        return Err(CommandError::Plugin(Error::PathForbidden(path)));
+    }
+
+    if fs_scope
+        .scope
+        .as_ref()
+        .map(|s| s.is_allowed(&path))
+        .unwrap_or(false)
+        || scope.is_allowed(&path)
+    {
         Ok(path)
     } else {
         Err(CommandError::Plugin(Error::PathForbidden(path)))
     }
 }
 
+fn is_forbidden<P: AsRef<Path>>(
```

Comment:
why is this implemented separately here?

Reply:
i can implement it in tauri::fs::Scope too and re-use that but wanted to keep the plugin compatible with all tauri v2 versions for now.

not too happy about the approach itself but merging `tauri::fs::Scope`s isn't possible rn so this was the next best thing to make the draft work

Reply:
let's ship with this for now, but I think we should still add it in `tauri` and then migrate the `fs` plugin later to that.

Reply:
ok, will open a pr later ~~if i don't forget~